flex-squid/src/main/java/org/sonar/flex/lexer/FlexLexer.java

```diff
@@ -43,36 +44,50 @@ private FlexLexer() {
 
   public static Lexer create(FlexConfiguration conf) {
     return Lexer.builder()
-        .withCharset(conf.getCharset())
+      .withCharset(conf.getCharset())
 
-        .withFailIfNoChannelToConsumeOneCharacter(true)
+      .withFailIfNoChannelToConsumeOneCharacter(true)
 
-        .withChannel(new BomCharacterChannel())
-        .withChannel(new BlackHoleChannel("\\s++"))
+      .withChannel(new BomCharacterChannel())
+      .withChannel(new BlackHoleChannel("\\s++"))
 
-        // Comments
-        .withChannel(commentRegexp("//[^\\n\\r]*+"))
-        .withChannel(commentRegexp("/\\*[\\s\\S]*?\\*/"))
+      // Comments
+      .withChannel(commentRegexp("//[^\\n\\r]*+"))
+      .withChannel(commentRegexp("/\\*[\\s\\S]*?\\*/"))
 
-        // String Literals
-        .withChannel(regexp(GenericTokenType.LITERAL, "\"([^\"\\\\]*+(\\\\[\\s\\S])?+)*+\""))
-        .withChannel(regexp(GenericTokenType.LITERAL, "\'([^\'\\\\]*+(\\\\[\\s\\S])?+)*+\'"))
+      // String Literals
+      .withChannel(regexp(GenericTokenType.LITERAL, "\"([^\"\\\\]*+(\\\\[\\s\\S])?+)*+\""))
+      .withChannel(regexp(GenericTokenType.LITERAL, "\'([^\'\\\\]*+(\\\\[\\s\\S])?+)*+\'"))
 
-        // Regular Expression Literal
-        .withChannel(new FlexRegularExpressionLiteralChannel())
+      // Regular Expression Literal
+      .withChannel(new FlexRegularExpressionLiteralChannel())
 
-        // Numbers
-        .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "0[xX][0-9a-fA-F]++"))
-        .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "[0-9]++\\.([0-9]++)?+" + EXP + "?+"))
-        .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "\\.[0-9]++" + EXP + "?+"))
-        .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "[0-9]++" + EXP + "?+"))
+      // Numbers
+      .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "0[xX][0-9a-fA-F]++"))
+      .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "[0-9]++\\.([0-9]++)?+" + EXP + "?+"))
+      .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "\\.[0-9]++" + EXP + "?+"))
+      .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "[0-9]++" + EXP + "?+"))
 
-        .withChannel(new IdentifierAndKeywordChannel("\\p{javaJavaIdentifierStart}++\\p{javaJavaIdentifierPart}*+", true, FlexKeyword.values()))
-        .withChannel(new PunctuatorChannel(FlexPunctuator.values()))
+      .withChannel(new IdentifierAndKeywordChannel("\\p{javaJavaIdentifierStart}++\\p{javaJavaIdentifierPart}*+", true, FlexKeyword.values()))
+      .withChannel(new PunctuatorChannel(FlexPunctuator.values()))
 
-        .withChannel(new UnknownCharacterChannel())
+      .withChannel(new UnknownCharacterChannel())
 
-        .build();
+      .build();
+  }
+
+  private static class BomCharacterChannel extends Channel<Lexer> {
```

Comment:
@vilchik-elena it's to properly set the column when there is a bom character, there is a failure at runtime with the old one on ruling on `sonar-flex/its/sources/src/as3-commons/as3-commons-asblocks/src/test/actionscript/org/as3commons/asblocks/test/AllTests.as`, with the old one the highlight starts at column 1, it has to start at column 0.